### PR TITLE
feat: Script supports installing kind and kubectl

### LINKED ISSUE #250: Script supports installing kind and kubectl
**What would you like to be added**:
Script supports installing kind and kubectl
**Why is this needed**:
Strive for a high degree of automation

### PR DESCRIPTION
**Pull request by lvtaococo**
#### What type of PR is this?
/kind feature

#### What does this PR do?

Script supports installing kind and kubectl.
#### Which issue(s) does this PR fix?

Fixes #250 
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: deploy/clustertree-cluster-manager.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ spec:
       containers:
         - name: clustertree-cluster-manager
           image: ghcr.io/kosmos-io/clustertree-cluster-manager:__VERSION__
-          imagePullPolicy: Always
+          imagePullPolicy: IfNotPresent
```
Comment (duanmengkk) [inline deploy/clustertree-cluster-manager.yml:63]: the verison is latest,so Always is better?

File: hack/install-go.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  install_go
+fi
+
+# Verify the Go version
+if ! go version | grep -q "go1.20"; then
+  echo "Installed Go version does not match the required version (1.20)."
+  install_go
+fi
+
+echo "Go is installed and the version is correct."
```
Comment (duanmengkk) [inline hack/install-go.sh]: add a new line

File: hack/install_kind_kubectl.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+BS_OS=$(go env GOOS)
+# check arch and os name before installing
+util::install_environment_check "${BS_ARCH}" "${BS_OS}"
+echo -n "Preparing: 'kubectl' existence check - "
+if util::cmd_exist kubectl; then
+  echo "passed"
+else
+  echo "not pass"
+  util::install_kubectl "" "${BS_ARCH}" "${BS_OS}"
+fi
```
Comment (duanmengkk) [inline hack/install_kind_kubectl.sh]: add a new line

File: deploy/clustertree-cluster-manager.yml (hunk 1 above)
Comment (lvtaococo) [reply deploy/clustertree-cluster-manager.yml:63]: The "latest" image hasn't been pushed to the remote repository.

File: hack/install-go.sh (hunk 2 above)
Comment (lvtaococo) [reply hack/install-go.sh]: ok

File: hack/install_kind_kubectl.sh (hunk 3 above)
Comment (lvtaococo) [reply hack/install_kind_kubectl.sh]: ok

**Pull request merged**